src/Psl/Xml/Issue/Level.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Issue;
+
+use Psl\Exception\InvariantViolationException;
+
+use function in_array;
+use function Psl\invariant;
+
+use const LIBXML_ERR_ERROR;
+use const LIBXML_ERR_FATAL;
+use const LIBXML_ERR_NONE;
+use const LIBXML_ERR_WARNING;
+
+/**
+ * @psalm-immutable
+ */
+final class Level
+{
+    private const LEVEL_ERROR = LIBXML_ERR_ERROR;
+    private const LEVEL_FATAL = LIBXML_ERR_FATAL;
+    private const LEVEL_NONE = LIBXML_ERR_NONE;
+    private const LEVEL_WARNING = LIBXML_ERR_WARNING;
+
+    private int $value;
+
+    private function __construct(int $value)
+    {
+        $this->value = $value;
+    }
+
+    /**
+     * @psalm-pure
+     * @throws InvariantViolationException
+     */
+    public static function fromLevelCode(int $value): self
+    {
+        invariant(
+            in_array($value, [self::LEVEL_ERROR, self::LEVEL_FATAL, self::LEVEL_NONE, self::LEVEL_WARNING], true),
+            'Received invalid XML issue level: ' . (string) $value
+        );
+
+        return new self($value);
+    }
```

Comment:
should be internal ( also in which case it will not throw invariant violation as it's in a controlled environment )

Reply:
... confused ... :)

What do you mean with should be internal?
It will always throw if the issue level is not know right?

Reply:
no, not really, if this is moved to `Psl\Xml\Internal\issue_level_from_xml(int $level): Level` we can be sure that we are always passing a valid $level so we don't need to check if it's one of XML errors.